zk_ee/src/common_structs/da_commitment_scheme.rs

```diff
@@ -0,0 +1,69 @@
+use crate::internal_error;
+use crate::oracle::usize_serialization::{UsizeDeserializable, UsizeSerializable};
+use crate::system::errors::internal::InternalError;
+
+///
+/// Rust representation of `L2DACommitmentScheme` from l1 contracts.
+/// It's used to define DA commitment zksync os outputs.
+///
+#[derive(Clone, Copy, Debug, PartialEq)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[repr(u8)]
+pub enum DACommitmentScheme {
+    /// Invalid option.
+    None,
+    /// Commitment will be equals to 0, used for validiums.
+    EmptyNoDA,
+    /// Keccak of stateDiffHash and keccak(pubdata). Can be used by custom DA solutions.
+    /// Currently not supported.
+    PubdataKeccak256,
+    /// This commitment includes EIP-4844 blobs data. Used by default RollupL1DAValidator.
+    /// With ZKsync OS it always outputs 1 0-hash blob, as separate commitment used for blobs.
+    BlobsAndPubdataKeccak256,
+    /// Keccak of blob versioned hashes filled with pubdata.
+    BlobsZKsyncOS,
+}
+
+impl TryFrom<u8> for DACommitmentScheme {
+    type Error = ();
+
+    fn try_from(value: u8) -> Result<Self, Self::Error> {
+        match value {
+            0 => Ok(DACommitmentScheme::None),
+            1 => Ok(DACommitmentScheme::EmptyNoDA),
+            2 => Ok(DACommitmentScheme::PubdataKeccak256),
+            3 => Ok(DACommitmentScheme::BlobsAndPubdataKeccak256),
+            4 => Ok(DACommitmentScheme::BlobsZKsyncOS),
+            _ => Err(()),
+        }
+    }
+}
+
+impl UsizeSerializable for DACommitmentScheme {
+    const USIZE_LEN: usize = <u8 as UsizeSerializable>::USIZE_LEN;
+
+    fn iter(&self) -> impl ExactSizeIterator<Item = usize> {
```

Comment:
Maybe it would be better to always explicitly convert it to/from `u8`? This code duplication here looks ugly

Reply:
> I don't think that this is a good reason honestly

I mean, this is the approach we are using usually in the codebase, even for a similar enum `TxEncodingFormat`. So I made the same to make it consistent.

Although this approach allows us to avoid intermediate types, I don't have a strict opinion on it, but IMHO, if we want to change it, it's worth changing it everywhere, and it's somewhat a separate issue.

Reply:
Also, to understand better: we don't want to move it into zksync-os-interfaces, right? Corresponding value is passed only by proof input generator?